lib/chef/resource/osx_profile.rb

```diff
@@ -308,19 +308,18 @@ def remove_profile
         #
 
         def get_installed_profiles(update = nil)
+          logger.trace("Saving profile data to node.run_state")
           if update
             node.run_state[:config_profiles] = query_installed_profiles
           else
             node.run_state[:config_profiles] ||= query_installed_profiles
           end
-          logger.trace("Saved profiles to run_state")
         end
```

Comment:
I feel like this function should be changed to return a value or cache the state, but that's probably outside the scope of fixing this for the next release.

Reply:
Also keep in mind this whole resource is basically EOL since it won't work on macOS 11. We need it to keep working for a few more years.